src/expr.rs

```diff
@@ -708,7 +708,7 @@ impl Rewrite for ast::Stmt {
                 };
 
                 let shape = shape.sub_width(suffix.len())?;
-                format_expr(ex, ExprType::Statement, context, shape).map(|s| s + suffix)
```

Comment:
I think that this is incorrect. We should only put the `if` expression in a single line if it is the last statement of the block.